scripts/find_outliers.py

```diff
@@ -22,6 +26,196 @@ def find_outliers(data_directory):
+            sqdiffs = np.sqrt(avgdiffs) # Return the square root of these values.
+            dvarsFirst.append(sqdiffs)
+        dvars_outside_brain[str(i)] = dvarsFirst
+        print('dvars ' + str(i) + ':' + str(len(dvars_outside_brain[str(i)])))
+
+#initially selecting and printing outliers from dvars that are 3 s.d. or more away from mean
+    outliers = {}
+    for i in range(len(dvars_outside_brain)):
+        num_vols = len(dvars_outside_brain[str(i)])
+        dvars_list = dvars_outside_brain[str(i)]
+        vol_outliers = []
+        for k in range(num_vols):
+            if dvars_list[k] > (np.mean(dvars_list) + 2.5*np.std(dvars_list)):
+                vol_outliers.append(k)
+            elif dvars_list[k] < (np.mean(dvars_list) - 2.5*np.std(dvars_list)):
+                vol_outliers.append(k)
+        outliers[str(i)] = vol_outliers
+
+    for i in range(len(outliers)):
+        print(str(imgfiles[str(i)]) + ' outliers:' + str(outliers[str(i)]))
+
+
+
+    # this function calculates the Dvars of the volumes
+    #dvars = {}
+
+    #for i in range(len(volumes)):
+        #nvoxels = volumes[str(i)].shape[0] * volumes[str(i)].shape[1] * volumes[str(i)].shape[2]
+        #num_vols = volumes[str(i)].shape[-1]
+        #dvarsFirst = []
+        #for elm in range(num_vols - 1):
+        #    data = volumes[str(i)]
+        #    diffs = data[:,:,:,elm] - data[:,:,:,elm + 1]
+        #    diffs = diffs**2 # Square the differences;
+        #    sumdiffs = sum(diffs.ravel()) # Sum over voxels for each volumevoxels;
+        #    avgdiffs = sumdiffs/nvoxels #divide by number of voxels
+        #    sqdiffs = np.sqrt(avgdiffs) # Return the square root of these values.
+        #    dvarsFirst.append(sqdiffs)
+        #dvars[str(i)] = dvarsFirst
+        #print(dvars[str(i)])
+
+    fig, ax = plt.subplots(20,1)
+    for i, ax in enumerate(ax):
+        dvars_run = dvars_outside_brain[str(i)]
+        ax.plot(dvars_run)
+        xs = outliers[str(i)]
+        #ys = dvars_run[outliers[str(i)]]
+        ys = [dvars_run[o] for o in outliers[str(i)]]
+        ax.scatter(xs, ys, edgecolor = None, color = 'r')
+        ax.set_ylabel('Run ' + str(i))
+        ax.set_xlim(0,162)
+    plt.show()
+
+
+            #return dvars
+        # Now here is where we do some dummy outlier code
+        #for volume in data(:,:,:,volumeidx)
+            # Do something mathematical to this volume
+            # such as find its mean amount and then
+            # put that value in an array for later comparison
+            # with the next future volumes
+            #volmath = [avgVal_vol1 avgVal_vol2 etc]
+
+        # if volume > criterion_for_being_outlier
+        #outlierarr[volume] = str[imgfile + '_' + volume]
+            #print(outlierarr)
+            #prints all vols that are outliers for this nifty file, then
+            #advances to the next nift file  named on the next line of
+            #"hast_list" '''
+
     raise RuntimeError('No code yet')
```

Comment:
Don't forget to remove the error - at the moment your code raises an error even though it has run correctly.